File: twitterEngine/src/spl/stomp/protocol/UnsubscribeFrame.java
package spl.stomp.protocol;

/**
 * @name Unsubscribe Frame
 * @author  Lidan Hifi
 * @author  Ken Saggy
 * 
 * UNSUBSCRIBE stomp frame
 */
class UnsubscribeFrame extends StompFrame {
    public UnsubscribeFrame() {
        super(StompCommand.UNSUBSCRIBE);
    }
    
    public UnsubscribeFrame(String id) {
        super(StompCommand.UNSUBSCRIBE);
        addFrameHeader("id", id);
    }
    
    @Override
    protected String[] getMandatoryHeaders() {
        return new String[] {"id"};
    }
}

File: twitterClient/twitterClient/src/stomp/frames/UnsubscribeFrame.cpp
//
//  UnsubscribeFrame.cpp
//  Object to represent the UNSUBSCRIBE frame type
//
//  Authors:    Ken Saggy
//              Lidan Hifi
//
//  Copyright 2013 Lidan Hifi, Ken Saggy

#include "../../../include/stomp/frames/UnsubscribeFrame.h"

namespace stomp {
    UnsubscribeFrame::UnsubscribeFrame(std::string identifier): StompFrame(UNSUBSCRIBE_FRAME_NAME) {
        addHeaderPair("id", identifier);
    }
    
    void UnsubscribeFrame::accept(FrameVisitor *visitor)
    { }
}

